lms/djangoapps/lti_provider/views.py

```diff
@@ -64,8 +66,11 @@ def lti_launch(request, course_id, usage_id):
     # Store the course, and usage ID in the session to prevent privilege
     # escalation if a staff member in one course tries to access material in
     # another.
-    params['course_id'] = course_id
-    params['usage_id'] = usage_id
+    course_key, usage_key = parse_course_and_usage_keys(course_id, usage_id)
+    if not course_key:
+        raise Http404('Invalid course or usage key')
```

Comment:
I've added the keys to the HTTP exception. Is it worth explicitly adding them to the log, since it's an error on the user's side?